resources/assets/js/volumes/components/imageGrid.vue

```diff
@@ -118,11 +118,11 @@ export default {
     computed: {
         columns() {
             // This might be 0 if the clientWidth is not yet initialized, so force 1.
-            return Math.max(1, Math.floor(this.clientWidth / (this.width + this.margin)));
+            return Math.max(1, Math.floor(this.clientWidth / (this.actualWidth + this.margin)));
         },
         rows() {
             // This might be 0 if the clientHeight is not yet initialized, so force 1.
-            return Math.max(1, Math.floor(this.clientHeight / (this.height + this.margin)));
+            return Math.max(1, Math.floor(this.clientHeight / (this.actualHeight + this.margin)));
```

Comment:
Since this already uses the margin, can't you do: `this.width + 2 * this.margin`? Then there is no need for a new computed property. Also, please add a comment why the margin is used twice here (because the CSS margin is 8 px and the border+padding is another 8 px).